### ISSUE TITLE
`Style/StringLiterals` does not work for `String`s interpolated into a `Regex`

### ISSUE THREAD

**Issue by vinnydiehl** [rubocop/rubocop#12061]
## To reproduce

With default settings, run RuboCop on:

```rb
/foo#{"sar".sub("s", "b")}/
```

## Expected behavior

A `Style/StringLiterals` warning, or maybe `Style/StringLiteralsInInterpolation`.

## Actual behavior

No warnings.

## RuboCop version

```
1.54.2 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
```

## Other thoughts

Would it be better to use `Style/StringLiteralsInInterpolation` here, rather than `Style/StringLiterals`? This would be consistent with other interpolations, however, the rule isn't quite as important in this case since there are no external `"` to cause confusion, as there would be with a `String` or `Symbol`.

**Comment by jonas054** [comment]
Yes, this seems to be a bug. We've overlooked interpolations inside regular expressions.

Since `Style/StringLiteralsInInterpolation` currently covers interpolations inside string literals and inside symbol literals, it seems obvious that it should handle regexp literals too.

> however, the rule isn't quite as important in this case since there are no external " to cause confusion

Agreed, but up until now we have not cared about this distinction.  `Style/StringLiteralsInInterpolation` reports when the wrong kind of quotes are used even when there are no outer quotes to cause confusion, e.g. in heredocs. So I say we should just add regexp literals to the duties of `Style/StringLiteralsInInterpolation` and not worry about which delimiters those outer literals use.
